gymnasium/envs/toy_text/frozen_lake.py

```diff
@@ -64,9 +65,13 @@ def generate_random_map(size: int = 8, p: float = 0.8) -> List[str]:
     valid = False
     board = []  # initialize to make pyright happy
 
+    rng = np.random
```

Comment:
Should this not be `np_random = seeding.np_random(seed)`

Reply:
I think this is meant to be essentially
```python
if seed is None:
    rng = np.random
else:
    rng, _ = seeding.np_random(seed)
```

The problem here is that `np_random` is annoyingly not an equivalent of `np.random` since `np.random` is essentially deprecated for this usage. Instead it should probably just be

```python
rng = np.default_rng(seed)
```

We might not need to use the whole `seeding.np_random` machinery, but if we want to, we can do `seeding.np_random(seed)` too, since it should handle `None` properly

Reply:
Yes, this is what I meant
The if statement can be simplified to `np_random = seeding.np_random(seed)` then `np_random.choice()`

Reply:
Updated as you suggested!